### ISSUE TITLE
TTS issues - Commander name issue

### ISSUE THREAD

**Issue by ghost** [EDCD/EDDI#1900]
## EDDI version in which issue is found
[Please give the full version number here, not just "latest" or "current"]
3.5.2 and 3.7.0

## VoiceAttack version in which issue is found (as applicable)
[Skip this section if VoiceAttack was not running when the issue occurred. Please give the full version number here, not just "latest" or "current"]

## Steps to reproduce
1. Enter a name in to the Commander Details -> Name (spoken) section
2. Click [Hear it]
3. Change the name to attempt to make the pronunciation clearer
4. Observe how this has not at all changed or updated
5. Uninstall EDDI and remove contents of %APPDATA% - EDDI - as noted in the troubleshooting section
6. Install version 3.7.0 of EDDI
7. Observe that a "blank" installation still has the stored spoken name in the TTS [Hear it] even though nothing is listed in the interface or the EDDI directory (although voice distortion and volume settings have been reset)

## Expected
To be able to update a commander name

## Observed
Only grabs initial commander name entry forever

## Investigation
As listed above, tried to perform a "clean installation" - without any of the other related programs running; VoiceAttack, E:D, Steam and EDMC were all terminated prior to these actions listed.

**Comment by Tkael** [comment]
It sounds like the default voice that you are using doesn't work very well with IPA in the speech synthesizer that EDDI uses. 
1. Which voice are you using?
2. What is the pronunciation that you are trying to set it to (so that we can attempt to replicate your settings)?
3. If you change to a different voice, does the IPA pronunciation work better?

**Comment by ghost** [comment]
The issue isn't the pronunciation, it is that it will not update. It took the initial entry and cannot be changed or removed. I just want to re-enter a new one. Even uninstalling "completely" - according to the troubleshooting section - it still keeps that setting.

I used the default voice and Microsoft Eva and a few others while testing. That made no difference.

The categorization you have now given this issue is incorrect.

**Comment by Darkcyde13** [comment]
I can confirm this.  I had entered my commander name as 'Darkcyde' some time ago, however somewhere along the line of upgrades, it seems that my commander name had been removed from the EDDI Commander Details page (it was blank when I looked just now), yet it is still using 'Darkcyde'.  When I click 'Hear it', or if I run the Honorific script, 'Darkcyde' is still used, regardless of what is set in the Commander details tab.  When I enter anything else, 'Darkcyde' is still used by EDDI.  This persists through EDDI restarts, i.e. set it as something new, restart EDDI, it still uses Darkcyde.

EDIT: I just tried through VA too, and it's the same.

**Comment by Tkael** [comment]
EDDI automatically takes the written commander name from the player journal (or from the Frontier API if it hasn't been set from the player journal and the Frontier API is configured. 

The **only thing that is configured** on the `Commander Details` tab **is the phonetic pronunciation** associated with the name. **The written name remains unchanged** when you modify the `Name (spoken)` field.

If the phonetic pronunciation fails (as it may for some voices - Ivona Amy for example often does not render IPA reliably) then the pronunciation will fall back to the written commander name. This could explain your observation that changing the field does not change the apparent output.

Another possibility is that you are using `{cmdr.name}` (which would be the written version of the name) rather than `{cmdr.phoneticname}` or `{CommanderName()}` (either of which would provide the phonetic rendering of that name with fallback to the written name if the phonetic rendering fails). (Note that neither is used in the default `Honorific` script...)

**Comment by ghost** [comment]
This is absolutely untrue, as the name it is parsing is not my commander name.
Also, why would there be a prompt for name entry when this is not what the program is using?
Why do you continually go back to a pronunciation issue when the data the program is using is at fault? If you would prefer not to locate the actual cause of the issue, then just cancel my request. I only tried to point out an issue.

**Comment by Darkcyde13** [comment]
Ah, my apologies, I misunderstood the description in the Commander Details tab.  It doesn't actually specify that EDDI gets this information from the Journal, so may I suggest a small update to that description to clarify?  Something like "EDDI obtains your commander name from the game Journal, however if the name of your commander is hard for EDDI to pronounce, then you can add a spoken version here, using ... [the rest of the existing description]".

**Comment by Tkael** [comment]
Apologies if I've misunderstood the issue. Text is at times a very imperfect medium.

**Comment by Darkcyde13** [comment]
I think I understand what gbobman means, well partially at least.

With my name, 'Darkcyde', Ivona Amy says it correctly.  If I change to MS Hazel, it is then pronounced as "Dark Sid".  I've tried to add a phonetic version in Commander Details, but nothing I add there makes any difference. (EDIT: to clarify, I mean when I click 'hear it').

gbobman's problem seems to be compounded by the fact that EDDI is getting a name that is not his in game commander name.  At least, that's how I understand his description.

EDIT2:  Even using {cmdr.phoneticname} in a blank script makes no difference, although I'm not sure if it's supposed to or not.

**Comment by Tkael** [comment]
`{cmdr.phoneticname}` or `{CommanderName()}` should render just like if you'd clicked the `Hear it` button.

**Comment by ghost** [comment]
Just a game, don't need to battle pride.

**Comment by Darkcyde13** [comment]
Not battling pride.  If there's a bug, or a misunderstanding, we (or at least I) would like to know either way.

That being said, even with dɑrkˈsaɪd entered in the Commander Details, it's still using the original 'Dark Sid' not the specified phonetic 'Darkside'.

**Comment by Tkael** [comment]
Odd. `dɑrkˈsaɪd` (as copied from your post and in Microsoft Hazel) seems to be working just fine for me (pronouncing the phonetic name like "Darkside" and definitely not like "Dark Sid", even though my written commander name is "Tkael").

**Comment by Darkcyde13** [comment]
Hmm, don't know what to say then.  I've tried Ivona Amy and MS Zira, both say my name properly anyway, without phonetics.  However, MS Hazel just refuses to pay any attention to what's entered in the Commander Details tab.

For me, this doesn't really matter, as I use Amy all the time, but if a similar thing is happening to others (like gbobman) I'd like to help if I can.  Is there anything else I can try for you Tkael?

**Comment by Tkael** [comment]
I'm baffled by the difference between what you're hearing and what I'm hearing when the conditions are so easy to replicate (just pasting the provided text into the field, using the same TTS voice, and clicking the `Hear it` button).

Maybe there's an issue with your copy of the voice? Might be worth uninstalling and reinstalling MS Hazel?

**Comment by Darkcyde13** [comment]
I just tried that, but turns out you can't uninstall default voice packages.  I even tried to install American English, then uninstall British English, but it's impossible.  I've looked online too, and those places that note such behaviour, do say you can't uninstall the defaults.  When I added American English, I'm also not able to make that my default, as English UK is the only one I can select.

I must admit, I've never been able to get the phonetics to work, but I've generally ignored it as Ivona Amy works well enough for me.  I've even fresh installed Win10 2004 recently, and that's made no difference either.

I did try enabling verbose logging when I tried last night, and in that there was a line that said what was being sent to the speech engine.  This was 'Darkcyde' exactly like my in-game name, so not the phonetic version of 'dɑrkˈsaɪd'.  Should it have done?

**Comment by Tkael** [comment]
Verbose logging should show the phonetic version. Do you have phonetic speech disabled in the Text-to-Speech tab?

**Comment by Darkcyde13** [comment]
I did check, just in case, but no, it's not ticked.

**Comment by Darkcyde13** [comment]
Weird, I just tried again, and now it works.  I've changed nothing within EDDI or Windows, so I'm baffled.  Here are the two logs.  I loaded EDDI and turned on verbose logging.  eddi.log is the one I tried just now where it works, and eddi3.log is the one I was trying last week where it failed.  (eddi1 and eddi2 logs were for sessions where I was testing a new VR headset, so are not relevant as I didn't make any changes to EDDI).
[EDDI logs.zip](https://github.com/EDCD/EDDI/files/4981788/EDDI.logs.zip)
The only thing I can think of that may have changed, is last weeks Windows updates, but I have no idea if there was anything in them that could have fixed it.

Well, anyway, it works now, so thanks for taking the time to look into this. :)

Oh, one thing I noticed in the eddi3.log is that it says I'm in Colonia, then says I've jumped to two other star systems, yet for that session I was in a station in Colonia and never moved.  Why would the log say I've jumped?
